### PR TITLE
HTML Traffic driver, tourist board and sponsor logo ad units

### PR DESCRIPTION
**Pull request by i-like-robots**
screen shot 2015-01-06 at 14 26 31

### CODE REVIEW

File: app/assets/stylesheets/core/layouts/modern/_ads.sass [hunk 1]
```diff
@@ -28,27 +28,42 @@
...
 
 .ad-leaderboard__inner
-  @extend %inline-block
-  display: inline-block
+  position: relative
+  display: table
+  margin-left: auto
+  margin-right: auto
+  +respond-to(728 + 180)
+    padding-left: 90px
```
Comment (Ianfeather) [inline app/assets/stylesheets/core/layouts/modern/_ads.sass]: What's the reason for this on top of the margins?

File: app/assets/stylesheets/core/layouts/modern/_ads.sass [hunk 2]
```diff
@@ -28,27 +28,42 @@
...
+
+.sponsor-logo
+  margin-top: 10px
+  +respond-to(728 + 180)
+    position: absolute
+    top: 0
+    left: 0
+    margin-top: 0
+
+.sponsor-logo__logo
```
Comment (Ianfeather) [inline app/assets/stylesheets/core/layouts/modern/_ads.sass]: mmm.... BEM

File: app/assets/stylesheets/core/layouts/modern/_ads.sass [hunk 3]
```diff
@@ -105,81 +120,34 @@
...
-// Page wallpaper
+// HTML Traffic Driver
 // ----------------------------------------------------------------------------
 
-.ad-oneByOne
-  display: none
+.traffic-driver
+  overflow: hidden
 
-.ad-wallpaper
```
Comment (Ianfeather) [inline app/assets/stylesheets/core/layouts/modern/_ads.sass]: +1

File: app/views/ads/html-sponsor-logo.html [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+      "<span class='sponsor-logo__badge badge badge--orange'>" + adLabelText + "</span>",
+    "</a>",
+  "</div>"
+].join("");
+
+if (adImpressionHref && /^(http|\/\/)/.test(adImpressionHref)) {
+  var impressionImg = document.createElement("img");
+  impressionImg.style.position = "absolute";
+  impressionImg.style.left = "-9999px";
+  impressionImg.src = adImpressionHref;
```
Comment (Ianfeather) [inline app/views/ads/html-sponsor-logo.html]: What's going on here? It seems a bit confusing, maybe a comment would be helpful. You're creating an image which sits off the page but never actually gets added to the page?

**Comment by Ianfeather** [comment]
Looks fine. Just a couple of questions

File: app/assets/stylesheets/core/layouts/modern/_ads.sass (hunk 1 above)
Comment (i-like-robots) [inline app/assets/stylesheets/core/layouts/modern/_ads.sass]: Making space for the logo while maintaining the central positioning of the leaderboard

File: app/views/ads/html-sponsor-logo.html (hunk 4 above)
Comment (i-like-robots) [inline app/views/ads/html-sponsor-logo.html]: I don't know what it does! I copied this from the old wallpaper ad template.

File: app/assets/stylesheets/core/layouts/modern/_ads.sass (hunk 1 above)
Comment (i-like-robots) [inline app/assets/stylesheets/core/layouts/modern/_ads.sass]: Admittedly I could use negative positioning also.

Comment (Ianfeather) [inline app/assets/stylesheets/core/layouts/modern/_ads.sass]: Cool, just didn't know there was a logo there

Comment (i-like-robots) [inline app/assets/stylesheets/core/layouts/modern/_ads.sass]: Screenshot added

**Comment by Ianfeather** [comment]
+1

**Pull request merged**